core/src/main/java/org/wildfly/extras/sunstone/api/impl/ec2/EC2CloudProvider.java

```diff
@@ -73,6 +74,22 @@ protected JCloudsNode createNodeInternal(String name, Map<String, String> overri
         return new EC2Node(this, name, overrides);
     }
 
+    @Override
+    protected String postProcessNodeGroupWhenCreatingNode(String nodeGroup) {
+        // JClouds place a limitation of 63 characters on the node group (see https://github.com/wildfly-extras/sunstone/issues/27)
```

Comment:
I have had a look through the JClouds code and I don't think the restriction applies to all providers.

I've been looking for usages of `FormatSharedNamesAndAppendUniqueStringToThoseWhichRepeat` and its `Factory` (the only place where `DnsNameValidator` is used). It seems to be instantiated in EC2, OpenStack and some other cloud providers we don't use.

I'd keep it specific for EC2, though it might make sense to implement the same check for OpenStack.